opensquirrel/passes/merger/general_merger.py

```diff
@@ -1,23 +1,25 @@
 from __future__ import annotations
 
+import inspect
 from abc import ABC, abstractmethod
 from math import acos, cos, floor, log10, sin
-from typing import cast
+from typing import Callable, cast
 
 import numpy as np
 
+from opensquirrel import I
 from opensquirrel.common import ATOL
-from opensquirrel.default_instructions import default_bloch_sphere_rotation_without_params_set
-from opensquirrel.ir import IR, Barrier, BlochSphereRotation, Instruction, Statement
+from opensquirrel.default_instructions import default_bloch_sphere_rotation_set
+from opensquirrel.ir import IR, Barrier, BlochSphereRotation, Float, Instruction, Statement
 from opensquirrel.utils import flatten_list
 
 
 def compose_bloch_sphere_rotations(a: BlochSphereRotation, b: BlochSphereRotation) -> BlochSphereRotation:
     """Computes the Bloch sphere rotation resulting from the composition of two Bloch sphere rotations.
     The first rotation is applied and then the second.
-    The resulting gate is anonymous except if `a` is the identity and `b` is not anonymous, or vice versa.
+    The resulting Bloch sphere rotation is always tried to be named, in case it were anonymous.
 
-    Uses Rodrigues' rotation formula, see for instance https://en.wikipedia.org/wiki/Rodrigues%27_rotation_formula.
+    Uses Rodrigues' rotation formula (see https://en.wikipedia.org/wiki/Rodrigues%27_rotation_formula).
     """
     if a.qubit != b.qubit:
         msg = "cannot merge two BlochSphereRotation's on different qubits"
```

Comment:
```suggestion
        msg = "cannot merge two BlochSphereRotations on different qubits"
```

Reply:
It's the Dutch `'s`, like in `foto's` 😀. I don't like writing `BlochSphereRotations` because that class doesn't exist. But I can write `two Bloch sphere rotations`.

Reply:
Maybe the latter is better here hahahahah